src/Adapter/SqlManager/RequestSqlDataProvider.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\SqlManager;
+
+use Db;
+use RequestSql;
+use Validate;
+
+/**
+ * Class RequestSqlDataProvider is responsible for providing data related to Request SQL model
+ */
+class RequestSqlDataProvider
+{
+    /**
+     * Get Request SQL data by given id
+     *
+     * @param $id
+     *
+     * @return array|null
+     */
+    public function getRequestSql($id)
+    {
+        if (!Validate::isLoadedObject($requestSql = new RequestSql($id))) {
+            return null;
+        }
+
+        return [
+            'id' => $requestSql->id,
+            'name' => $requestSql->name,
+            'sql' => $requestSql->sql,
+        ];
+    }
+
+    /**
+     * Get all database tables
+     *
+     * @return array
+     */
+    public function getTables()
+    {
+        return (new RequestSql())->getTables();
```

Comment:
If we use  `getTables` or `getTableColumns` twice `new RequestSql` is rebuild every time. Maybe instanciate it just once and use it when needed